code/game/objects/objs.dm

```diff
@@ -314,8 +314,14 @@ a {
 /obj/proc/container_resist(var/mob/living)
 	return
 
-/obj/proc/CanAStarPass()
+/obj/proc/CanAStarPass(ID, dir, caller)
 	. = !density
+	if(!. && istype(caller, /mob/living/simple_animal/parrot)) // Only check below if we can't pass to save up on calls
```

Comment:
This is pretty gross and reallyyy snowflakey as far as code goes.

Reply:
I agree but this part is necessary to make AStar consider perches as valid nodes. What if I flipped that and made `/atom/proc/CanAStarPassTo(ID, dir, destination) ` (could even be `mob`) that returns whether the source should consider `destination` as a good node? Then when checking the links during AStar I could make an or that checks if either `CanAStarPass` or `CanAStarPassTo` return `TRUE` then the object is a valid node.